cylc/uiserver/app.py

```diff
@@ -128,16 +122,18 @@ class CylcUIServer(ExtensionApp):
     config_file_paths = list(
         map(
             str,
-            [
-                # user configuration
-                USER_CONF_ROOT,
-                # site configuration
-                SITE_CONF_ROOT,
-                # base configuration - always used
-                Path(uis_pkg).parent,
-            ]
+            get_conf_dir_hierarchy(
+                [
+                    # user configuration
+                    USER_CONF_ROOT,
+                    # site configuration
+                    SITE_CONF_ROOT,
+                ], filename=False
+            )
         )
     )
+    config_file_paths.append(str(Path(uis_pkg).parent))
```

Comment:
We should add the path hierarchy here rather than loading the hub config, this lets Jupyter Server take care of loading the files correctly.